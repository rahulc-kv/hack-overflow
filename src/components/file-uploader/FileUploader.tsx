import React, { FC } from 'react';

import { UploadIcon } from '@assets/icons';
import { FileUploaderProps } from './types';

const FileUploader: FC<FileUploaderProps> = props => {
  const { label, name, acceptedFormats = '', handleOnChange } = props;

  return (
    <>
      <input
        id={name}
        type="file"
        name={name}
        accept={acceptedFormats}
        className="overflow-hidden absolute w-0 h-0 opacity-0"
        onChange={handleOnChange}
        data-testid="fileUploaderInput"
      />
      <label htmlFor={name} className="cursor-pointer">
        <div
          className="flex flex-row justify-center items-center py-[7px] px-[17px]
           text-sm text-white bg-primaryColor rounded-md 2xl:px-[65px]">
          <UploadIcon width={24} height={24} className="mr-[9px]" />
          {label}
        </div>
      </label>
    </>
  );
};

export default FileUploader;
