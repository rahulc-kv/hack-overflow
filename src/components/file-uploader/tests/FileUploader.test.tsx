import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import FileUploader from '../FileUploader';

const handleOnChange = jest.fn();

test('FileUploader rendered', () => {
  const component = render(
    <FileUploader
      label={'Test'}
      name={'Test'}
      acceptedFormats=""
      handleOnChange={handleOnChange}
    />
  );
  expect(component).toMatchSnapshot();
});

test('FileUploader Click', () => {
  const { getByTestId } = render(
    <FileUploader
      label={'Test'}
      name={'Test'}
      acceptedFormats=""
      handleOnChange={handleOnChange}
    />
  );
  fireEvent.change(getByTestId('fileUploaderInput'), { target: { value: '' } });
  expect(handleOnChange).toBeCalled();
});
