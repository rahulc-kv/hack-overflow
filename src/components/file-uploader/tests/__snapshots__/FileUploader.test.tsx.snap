// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        accept=""
        class="overflow-hidden absolute w-0 h-0 opacity-0"
        data-testid="fileUploaderInput"
        id="Test"
        name="Test"
        type="file"
      />
      <label
        class="cursor-pointer"
        for="Test"
      >
        <div
          class="flex flex-row justify-center items-center py-[7px] px-[17px] text-sm text-white bg-primaryColor rounded-md 2xl:px-[65px]"
        >
          <svg
            class="mr-[9px]"
            height="24"
            width="24"
          >
            uploadIcon.svg
          </svg>
          Test
        </div>
      </label>
    </div>
  </body>,
  "container": <div>
    <input
      accept=""
      class="overflow-hidden absolute w-0 h-0 opacity-0"
      data-testid="fileUploaderInput"
      id="Test"
      name="Test"
      type="file"
    />
    <label
      class="cursor-pointer"
      for="Test"
    >
      <div
        class="flex flex-row justify-center items-center py-[7px] px-[17px] text-sm text-white bg-primaryColor rounded-md 2xl:px-[65px]"
      >
        <svg
          class="mr-[9px]"
          height="24"
          width="24"
        >
          uploadIcon.svg
        </svg>
        Test
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
